frontend/functions/authorFunctions.ts

```diff
@@ -1,31 +1,38 @@
-import {
-  Author, Mandate, MeHeaderQuery, Member,
+import { i18n } from 'next-i18next';
+import selectTranslation from '~/functions/selectTranslation';
+import
+{
+  ArticleQuery,
+  MeHeaderQuery, Member,
 } from '~/generated/graphql';
 import { getFullName } from './memberFunctions';
 
+export const getMemberSignature = (member: Member): string => getFullName(member);
 // eslint-disable-next-line import/prefer-default-export
-export const getSignature = (author: Partial<Author>): string => {
-  if (author.__typename === 'Member') {
-    return getFullName(author as Member);
+export const getSignature = (author: ArticleQuery['article']['author']): string => {
+  if (author.type === 'Custom' && author.customAuthor) {
+    return selectTranslation(i18n, author.customAuthor.name, author.customAuthor.nameEn);
   }
-  const mandate = (author as Mandate);
-  if (mandate.position?.id?.startsWith('dsek.noll.stab')) {
-    return 'Staben';
+  if (author.type === 'Mandate' && author.mandate && author.mandate.position?.name) {
+    return `${getFullName(author.member)}, ${author.mandate.position.name}`;
   }
-
-  return `${getFullName(mandate.member)}, ${mandate.position.name}`;
+  return getMemberSignature(author.member);
 };
 
-export const getAuthor = (author: Partial<Author>): Partial<Member> => {
-  if (author.__typename === 'Member') {
-    const member = author as Member;
-    return member;
+export const getAuthorImage = (author: ArticleQuery['article']['author']): string | undefined => {
+  if (author.type === 'Custom') {
+    if (!author.customAuthor.imageUrl) return undefined;
+    return author.customAuthor.imageUrl;
   }
-  const mandate = author as Mandate;
-  return mandate.member;
+  return author.member?.picture_path || '';
 };
 
-export const getAuthorId = (author: Partial<Author>): string => getAuthor(author).id;
-export const getAuthorStudentId = (author: Partial<Author>): string => getAuthor(author).student_id;
-
-export const authorIsUser = (author: any, user: MeHeaderQuery['me']) => author && user && getAuthorId(author) === user.id;
+export const authorIsUser = (author: ArticleQuery['article']['author'], user: MeHeaderQuery['me']) => {
+  if (!user) return false;
+  if (author?.member?.id === user?.id) return true;
```

Comment:
Can we add an if-statement checking if the user has the author.mandate (if it exists)?

Reply:
Since mandate is directly connected to a member, the author.member.id will be the same as the author.mandate.member.id (if it exists).

Reply:
Aha of course.